src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/AttachDocsToSupplementaryEvidence.java

```diff
@@ -48,37 +48,48 @@ public void attach(Envelope envelope, CaseDetails existingCase) {
                 existingCase.getCaseTypeId()
             );
 
-            CcdAuthenticator authenticator = ccdApi.authenticateJurisdiction(envelope.jurisdiction);
+            try {
+                CcdAuthenticator authenticator = ccdApi.authenticateJurisdiction(envelope.jurisdiction);
 
-            StartEventResponse startEventResp = ccdApi.startEvent(
-                authenticator,
-                envelope.jurisdiction,
-                existingCase.getCaseTypeId(),
-                existingCase.getId().toString(),
-                EVENT_TYPE_ID
-            );
+                StartEventResponse startEventResp = ccdApi.startEventForExistingCase(
+                    authenticator,
+                    envelope.jurisdiction,
+                    existingCase.getCaseTypeId(),
+                    existingCase.getId().toString(),
+                    EVENT_TYPE_ID
+                );
 
-            log.info(
-                "Started event in CCD for envelope ID: {}. File name: {}. Case ref: {}",
-                envelope.id,
-                envelope.zipFileName,
-                existingCase.getId()
-            );
+                log.info(
+                    "Started event in CCD for envelope ID: {}. File name: {}. Case ref: {}",
+                    envelope.id,
+                    envelope.zipFileName,
+                    existingCase.getId()
+                );
 
-            ccdApi.submitEventForExistingCase(
-                authenticator,
-                envelope.jurisdiction,
-                existingCase.getCaseTypeId(),
-                existingCase.getId().toString(),
-                buildCaseDataContent(envelope, startEventResp)
-            );
+                ccdApi.submitEventForExistingCase(
+                    authenticator,
+                    envelope.jurisdiction,
+                    existingCase.getCaseTypeId(),
+                    existingCase.getId().toString(),
+                    buildCaseDataContent(envelope, startEventResp)
+                );
 
-            log.info(
-                "Attached documents from envelope to case. Case ID: {}, envelope ID: {}",
-                existingCase.getId(),
-                envelope.id
-            );
+                log.info(
+                    "Attached documents from envelope to case. Case ID: {}, envelope ID: {}",
+                    existingCase.getId(),
+                    envelope.id
+                );
+            } catch (UnableToAttachDocumentsException e) {
+                log.error(
+                    "Failed to attach documents from envelope to case. Case Type: {}, Case ID: {}, envelope ID: {}",
+                    existingCase.getCaseTypeId(),
+                    existingCase.getId(),
+                    envelope.id
+                );
+                return false;
```

Comment:
is there a benefit of replacing throwing an exception with boolean response?
imho it would be clearer tho handle this exception where this method is called.

Reply:
There's some elegance in this solution. If we don't need to control the flow using exceptions, I think best not to. Using this method is quite straightforward as it is. As long as it returns a result, we always need to continue the processing of the message as if nothing bad happened. If it throws, that's a message processing error. Not that handling this exception in the calling class would be something incorrect, but I find this way a bit nicer - it leaves the calling code very clear.

It would be worth explaining in javadoc what is returned here.